pkgs/flox-buildenv/default.nix

```diff
@@ -0,0 +1,67 @@
+{
+  cacert,
+  darwin,
+  coreutils,
+  flox-activation-scripts,
+  flox-pkgdb,
+  glibcLocalesUtf8,
+  lib,
+  nix,
+  perl,
+  runCommandNoCC,
+  stdenv,
+  writers,
+  writeText,
+}:
+let
+  pname = "flox-buildenv";
+  version = "0.0.1";
+  buildenv = (writers.writeBash "buildenv" (builtins.readFile ../../buildenv/buildenv.bash));
+  buildenv_nix = ../../buildenv/buildenv.nix;
+  builder_pl = ../../buildenv/builder.pl;
+  activationScripts = flox-activation-scripts;
+  defaultEnvrc = writeText "default.envrc" (
+    ''
+      # Default environment variables
+      export SSL_CERT_FILE="''${SSL_CERT_FILE:-${cacert}/etc/ssl/certs/ca-bundle.crt}"
+      export NIX_SSL_CERT_FILE="''${NIX_SSL_CERT_FILE:-''${SSL_CERT_FILE}}"
+    ''
+    + lib.optionalString stdenv.isLinux ''
+      export LOCALE_ARCHIVE="''${LOCALE_ARCHIVE:-${glibcLocalesUtf8}/lib/locale/locale-archive}"
+    ''
+    + lib.optionalString stdenv.isDarwin ''
+      export NIX_COREFOUNDATION_RPATH="''${NIX_COREFOUNDATION_RPATH:-"${darwin.CF}/Library/Frameworks"}"
+      export PATH_LOCALE="''${PATH_LOCALE:-${darwin.locale}/share/locale}"
+    ''
+    + ''
+      # Static environment variables
+    ''
+  );
```

Comment:
suggestion: define separately or as a `passthru`.

currently this is inaccessible when calling the bare `buildenv.nix` as we would from the cli.

Reply:
I'd be happy to entertain a diff but I'm not sure what you're specifically suggesting here. Not even sure what you mean by "this"? The `defaultEnvrc` derivation?

Reply:
yes, if we call `nix build <buildenv.nix> --argstr defaultEnvrc <path to defaultEnvrc> ...`, we need to know `<path to defaultEnvrc>` in the cli but as it is now its entirely opaque.

An alternative for these is always to build the buildenv package first before every run of flox (in dev, e.g through just) similar to how pkgdb is built separately and then used from a dynamic path at runtime.
Its a bit of a pain because we end up setting overrides all over the place shell hooks, the justfile, flox-cli-tests.
It can also lead to some annoying delay when running `just build` when there is a number of builds happening before flox is started.

If that is what we want for these as well, i'd suggest that we build the entire `buildenv.nix` to a gcroot (so that running `flox` can use that as well) and add a justfile rule to rebuild that upon `just build` (and do the same for the manifest builder i think).

That way at least we only have compile a single path into the cli.

Reply:
> we need to know <path to defaultEnvrc> in the cli but as it is now its entirely opaque.

The `defaultEnvrc` file is built using nixpkgs paths, and while it's possible to provide those paths to the CLI and have it create and pass the file to the derivation that's kinda silly because the CLI has no other use for those values. By creating it here as close as possible to where it is used you have much better data locality and less complexity in the CLI build.

I'm anxious not to let the dev env be the tail that wags the dog, and as discussed in the other conversation we did manage to solve this for pkgdb, albeit using make. If the same trick can be performed using just then I'd be happy to see that happen, I just wouldn't be at all quick to do that work.

Reply:
Spoke with Yannik and confirmed that the default envrc isn't something that needs to be passed down from CLI, but that we will want to update the justfile to rebuild the buildenv package automatically when files change. So resolving conversation for now but will implement this in the followup PR where we call the new buildenv from rust.